### PR TITLE
Add MySQL flexible server configuration

### LINKED ISSUE #1633: Support for Azure Database for MySQL Flexible Server
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

It doesn't seem like the https://www.pulumi.com/registry/packages/azure-native/api-docs/dbformysql/ API allows us to create flexible servers. This feels similar to #1455 but I'm not sure if the APIs for postgres and mysql are different enough that this one was missed. Nothing in the documentation is mentioning flexible servers so I'm assuming something is missing.

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Fix #1633

Adds an override to enable generation of the `Configuration` resource for MySQL flexible servers. Flexible servers are available already, but users have to use explicit API namespaces. To make it easier, I also added a usage example to refer to.

### CODE REVIEW

File: versions/spec-resources.json [hunk 1]
```diff
@@ -3959,7 +3959,12 @@
     ],
     "Configuration": [
       "2017-12-01",
-      "2017-12-01-preview"
+      "2017-12-01-preview",
```
Comment (thomas11) [inline versions/spec-resources.json:3962]: These two files are generated, not manually edited unless I'm not aware of an exception. `make versions` will overwrite these changes.

Comment (mikhailshilkov) [reply versions/spec-resources.json:3962]: I haven't edited them manually, they were auto-updated

Comment (thomas11) [reply versions/spec-resources.json:3962]: Oh, I was led astray by the lack of a change in v1-config.yaml, but that's not the only input to these files.

**Review by danielrbradley — APPROVED** [review]
Looks good :shipit:

**Pull request merged**